mesonbuild/compilers/mixins/gnu.py

```diff
@@ -306,7 +306,9 @@ def get_include_args(self, path: str, is_system: bool) -> T.List[str]:
 
     @classmethod
     def use_linker_args(cls, linker: str) -> T.List[str]:
-        if linker not in {'gold', 'bfd', 'lld'}:
+        if linker not in {'gold', 'bfd', 'lld',
+                # qcld: Qualcomm Snapdragon linker, based on LLVM.
+                'qcld'}:
```

Comment:
Comment added here to describe what is 'qcld'.

Reply:
Does this linker actually work with gnu? I think we probably want to put this in the clang implementation.

Reply:
Oh. That explains why Meson (w/o my patch) classified the Qualcomm linker as Apple's clang. I am happy to rewrite the patch if you can pass me some pseudo code of the derived class.

Reply:
in https://github.com/mesonbuild/meson/blob/master/mesonbuild/compilers/mixins/clang.py#L111 you want to add something like:

```python
if linker == 'qcld':
    return ['-fuse-ld=qcld']
```